From 78ed1ec669a54690114be75635c21ce5a84cc573 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Mon, 5 Jan 2015 18:45:29 +0800
Subject: [PATCH 559/635] MA-6065 - [Android_MX6DQ/DL SABRE-AI Board]Taking
 photos:take one or two

pictures,then the keys of taking photos cannot press

In camera framework(RequestThreadManager.java), if no jpeg data call back
in 3 seconds, will cause this issue.

Uvc need do csc, the resolution 2304x1536 cause too much time.
So just omit large resolution for uvc.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 mx6/libcamera2/UvcDevice.cpp | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/mx6/libcamera2/UvcDevice.cpp b/mx6/libcamera2/UvcDevice.cpp
index 2c77c2a..3fc8692 100644
--- a/mx6/libcamera2/UvcDevice.cpp
+++ b/mx6/libcamera2/UvcDevice.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2012-2014 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2015 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -226,12 +226,14 @@ status_t UvcDevice::initSensorInfo(const CameraInfo& info)
                     convertPixelFormatToV4L2Format(mDefaultFormat);
         ret = ioctl(mCameraHandle,
                     VIDIOC_ENUM_FRAMESIZES, &vid_frmsize);
-        if (vid_frmsize.discrete.width == 1600 &&
-                     vid_frmsize.discrete.height == 1200) {
-            continue;
-        }
 
         if (ret == 0) {
+            //uvc need do csc, so omit large resolution.
+            if (vid_frmsize.discrete.width > 1920 ||
+                     vid_frmsize.discrete.height > 1080) {
+                continue;
+            }
+
             FLOGI("enum frame size w:%d, h:%d",
                        vid_frmsize.discrete.width, vid_frmsize.discrete.height);
             memset(&vid_frmval, 0, sizeof(struct v4l2_frmivalenum));
-- 
1.8.0

